dev-docs/specs/project-workspaces.md

```diff
@@ -0,0 +1,441 @@
+# Workspace Architecture Implementation Plan
+
+## Overview
+Refactor the storage system to use a unified workspace model where all storage is organized as workspaces containing projects. Add breadcrumb UI showing "workspace / project", improve import flow with project naming, and add workspace naming on open.
+
+## Key Design Decisions
+- **Workspace Structure**: Flat - `workspace-folder/project-name/noodles.json + data/`
+- **Built-in Workspaces**: Browser Storage (OPFS) and Examples (publicFolder) act as special workspaces
+- **Import Flow**: Import saves to current workspace, prompting for project name (prefilled with basename)
+- **Workspace Naming**: Always prompt when opening a folder workspace; name becomes the identifier
+- **New Project**: Always prompts for workspace selection first if none is active
+- **URL Format**: `?workspace=name&project=name`
```

Comment:
Could this use paths like in the Wouter PR?

Say /examples/nyc-taxis vs /projects/nyc-taxis?